tajo-core/src/main/java/org/apache/tajo/querymaster/Query.java

```diff
@@ -402,6 +403,39 @@ public QueryState getSynchronizedState() {
     }
   }
 
+  public boolean hasUnionPlan() throws Exception {
+    boolean result = false;
+
+    // During executing UNION statement or UNION ALL statement, the terminal block doesn't have join plan and has
+    // two more child blocks. In above case, Tajo should summarize result TableStats from all child stages.
+    ExecutionBlock terminalBlock = plan.getTerminalBlock();
+    terminalBlock.setPlan(terminalBlock.getPlan());
+    if (plan.getChilds(terminalBlock.getId()).size() >= 2) {
+      result = true;
+    }
+    return result;
+  }
+
+  public TableStats computeChildResultStats() throws Exception {
```

Comment:
A better implementation exists at Stage.computeStatFromUnionBlock(). It would be better to move this method to StatisticsUtil.